modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/NameSign.java

```diff
@@ -15,49 +14,50 @@
 import org.jetbrains.annotations.NotNull;
 
 public final class NameSign implements Listener {
-    private static final String HEADER = "Name:";
+
+    private static final String NAME_SIGN_IDENTIFIER = "Name:"; //String on first line which will identify the block as a name sign
+
+    /**
+     * When a craft is piloted, check the craft for a name sign
+     */
     @EventHandler
     public void onCraftDetect(@NotNull CraftDetectEvent event) {
-        Craft c = event.getCraft();
+        Craft craft = event.getCraft();
 
-        if(c.getNotificationPlayer() == null || (Settings.RequireNamePerm && !c.getNotificationPlayer().hasPermission("movecraft.name.use"))) {
-            //Player is null or does not have permission (when required)
+        //Ensure nonnull player and sufficient permissions
+        if (craft.getNotificationPlayer() == null || (Settings.RequireNamePerm && !craft.getNotificationPlayer().hasPermission("movecraft.name.use"))) {
             return;
         }
 
-        World w = c.getW();
+        World craftWorld = craft.getW();
+        for (MovecraftLocation location : craft.getHitBox()) { //Iterate through every craft block to search for name sign
+            Block b = location.toBukkit(craftWorld).getBlock();
```

Comment:
if you're trying to be more explicit in your variable naming, than `b` should be `block`. You switch between 3 naming conventions (`World craftWorld`, `Block b`, `Craft craft`).

Reply:
I'd prefer `Block block`, `Craft craft` and `World world`